### PR TITLE
Remove ENABLE_DENOISER definition, create denoiser on demand

### PR DESCRIPTION
**Pull request by dtarakanov1**

### CODE REVIEW

File: BaikalStandalone/Application/app_utils.h [hunk 1]
```diff
@@ -102,11 +102,9 @@ namespace Baikal
         float gpu_mem_fraction = 0; // float number from 0 to 1, percentage of max used device memory, 0 for default behavior
         std::string visible_devices;
 
-#ifdef ENABLE_DENOISER
         // denoiser settings
-        DenoiserMode denoiser_type = DenoiserMode::kML;
-        std::uint32_t start_spp;
-#endif
+        DenoiserMode denoiser_type = DenoiserMode::kNone;
+        std::uint32_t start_spp = 8;
```
Comment (achernigin1987) [inline BaikalStandalone/Application/app_utils.h]: denoiser_start_spp ?

File: Baikal/Utils/output_accessor.h [hunk 2]
```diff
@@ -22,6 +22,28 @@ namespace Baikal
...
+                for (auto& output : outputs[device_idx])
+                {
+                    SaveImageFromRendererOutput(
+                        device_idx,
+                        output.first,
+                        output.second.get(),
+                        m_frame_count);
+                }
+            }
+        }
```
Comment (AvKhokhlov) [inline Baikal/Utils/output_accessor.h]: Do we really need NextFrame here?

Comment (AvKhokhlov) [reply Baikal/Utils/output_accessor.h]: Move it into BaikalStandalone

File: Baikal/RenderFactory/render_factory.h [hunk 3]
```diff
@@ -35,6 +35,7 @@ namespace Baikal
 
     enum class PostEffectType
     {
+        kNone,
```
Comment (achernigin1987) [inline Baikal/RenderFactory/render_factory.h]: Why we need kNone here ?

Comment (dtarakanov1) [reply Baikal/RenderFactory/render_factory.h]: It is to distinguish cases where the `denoiser_type` parameter is not specified.

Comment (achernigin1987) [reply Baikal/RenderFactory/render_factory.h]: DenoiserMode::kNone is enough

File: BaikalStandalone/Application/cl_render.cpp [hunk 4]
```diff
@@ -23,6 +23,7 @@ THE SOFTWARE.
 
 #include "Application/cl_render.h"
 #include "Application/gl_render.h"
+#include "Utils/output_accessor.h"
```
Comment (achernigin1987) [inline BaikalStandalone/Application/cl_render.cpp]: #ifdef DUMP...

File: BaikalStandalone/Application/cl_render.cpp [hunk 5]
```diff
@@ -52,40 +49,43 @@ namespace Baikal
...
 
     AppClRender::AppClRender(AppSettings& settings, GLuint tex)
-    : m_tex(tex), m_output_type(Renderer::OutputType::kColor), m_frame_count(0)
+    : m_tex(tex)
     {
         InitCl(settings, m_tex);
 
-#ifdef ENABLE_DENOISER
-        switch (settings.denoiser_type)
+        m_denoiser_type = settings.denoiser_type;
```
Comment (achernigin1987) [inline BaikalStandalone/Application/cl_render.cpp]: May be after "switch" ? If exception will be raised and client will catch exception then m_denoiser_type will be wrong

File: BaikalStandalone/Utils/output_accessor.h [hunk 6]
```diff
@@ -145,7 +164,9 @@ namespace Baikal
         std::vector<RadeonRays::float3> m_image_data;
         std::vector<RGB> m_image_rgb;
         std::string m_output_dir;
-        size_t m_width;
-        size_t m_height;
+        std::size_t m_width;
+        std::size_t m_height;
+        std::size_t m_frame_count = 0;
```
Comment (achernigin1987) [inline BaikalStandalone/Utils/output_accessor.h]: may be static ?

File: BaikalStandalone/Application/cl_render.cpp (hunk 4 above)
Comment (dtarakanov1) [reply BaikalStandalone/Application/cl_render.cpp]: Here it does not make difference.

**Pull request merged**